var/spack/repos/builtin/packages/blaspp/package.py

```diff
@@ -53,12 +53,18 @@ def cmake_args(self):
             if '+rocm' in spec:
                 backend = 'hip'
             backend_config = '-Dgpu_backend=%s' % backend
+
+        args = ''
+        if (spec['blas'].name == 'cray-libsci'):
+            args+='-DBLA_VENDOR=CRAY '
+
         return [
             '-Dbuild_tests=%s'       % self.run_tests,
             '-Duse_openmp=%s'        % ('+openmp' in spec),
             '-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec),
             backend_config,
-            '-DBLAS_LIBRARIES=%s'    % spec['blas'].libs.joined(';')
+            '-DBLAS_LIBRARIES=%s'    % spec['blas'].libs.joined(';'),
+            args
```

Comment:
This is not right, and you'll also note the style tests are failing due to lines 59 and probably for the superfluous parens in 58. Do:
```
        args = [
            '-Dbuild_tests=%s'       % self.run_tests,
            '-Duse_openmp=%s'        % ('+openmp' in spec),
            '-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec),
            backend_config,
            '-DBLAS_LIBRARIES=%s'    % spec['blas'].libs.joined(';'),
        ]

        if spec['blas'].name == 'cray-libsci':
            args.append(self.define('BLA_VENDOR', 'CRAY'))

        return args
```